docs/src/content/docs/reference/configuration.mdx

```diff
@@ -444,6 +444,11 @@ starlight({
 });
 ```
 
+:::caution
+Scripts are placed into head using the `set:html` directive which is similar to using `is:inline` in that it [precludes these scripts from being processed by Astro](https://docs.astro.build/en/guides/client-side-scripts/#opting-out-of-processing).
+This means that if you want to include a local script and you need this to be processed, you'll probably need to [override or extend the Head component](/guides/overriding-components/#reuse-a-built-in-component).
+:::
```

Comment:
Instead of a caution, I think we can add a clear description of how this _does_ work (rather than specifying how it doesn’t):

```suggestion
Entries in `head` are converted directly to HTML elements and do not pass through Astro’s script or style processing.
If you need to import local assets like scripts, styles, or images, [override the Head component](/guides/overriding-components/#reuse-a-built-in-component).
```

This also helps us make this a bit simpler because we don’t need to explain too much all these other things which `head` _isn’t_, instead focusing on the simpler case of what it is.

Reply:
Done, I did embed a link to the Astro style and script docs in case users reading this will want to learn what this style and script processing is all about